### PR TITLE
Feature/competition

### PR DESCRIPTION
**Pull request by patrikulus**
Migrated Wroc# competition app from https://github.com/rmaziarka/WrocSharpCompetition to .NET Core 2.1

## Description
What has been done:
* Created new clean ASP.NET Core MVC solution and moved code from source app
* Refactored all namespaces 
* Refactored some parts of code to fit .NET Core
* Commented problematic parts and marked with TODO
* Removed custom filter forcing https because it will be covered by rule on CloudFlare

## Where should the reviewer start?
I think the only reasonable review in this case is to compare solution structure with base one and verify if all required code has been moved. 

## Motivation and context
This is our code base for new competition app.
#1 

## How has this been tested?
App is compiling and starts without errors.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by patrikulus** [comment]
[Przepisanie aplikacji od Radka na .NET Core](https://trello.com/c/XpUfSTcJ/37-przepisanie-aplikacji-od-radka-na-net-core)

**Comment by WTobor** [comment]
[Przepisanie aplikacji Wroc# na .NET Core](https://trello.com/c/1aPxbz0O/1-przepisanie-aplikacji-wroc-na-net-core)

File: src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Views/Account/ConfirmEmail.cshtml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿@{
+    ViewData["Title"] = "Confirm email";
+}
+
+<h2>@ViewData["Title"]</h2>
+<div>
+    <p>
+        Thank you for confirming your email.
```
Comment (WTobor) [inline src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Views/Account/ConfirmEmail.cshtml]: why there is missing `Please @Html.ActionLink("Click here to Log in", "Login", "Account", routeValues: null, htmlAttributes: new { id = "loginLink" })` part?

File: src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Views/Account/ExternalLogin.cshtml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+﻿@model ExternalLoginViewModel
```
Comment (WTobor) [inline src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Views/Account/ExternalLogin.cshtml]: Is it duplicated _ExternalLoginConfirmation.cshtml_?

File: src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Views/Account/Register.cshtml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+﻿@model RegisterViewModel
+@{
+    ViewData["Title"] = "Register";
+}
+
+<h2>@ViewData["Title"]</h2>
+
+<div class="row">
+    <div class="col-md-4">
+        <form asp-route-returnUrl="@ViewData["ReturnUrl"]" method="post">
```
Comment (WTobor) [inline src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Views/Account/Register.cshtml]: Why we don't have _FirstName_ and _SecondName_? Or just _UserName_?

File: src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Controllers/HomeController.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using DevAdventCalendarCompetition.Models;
+
+namespace DevAdventCalendarCompetition.Controllers
+{
+    public class HomeController : Controller
```
[the remark this answers is not in the document]
Comment (patrikulus) [reply src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Controllers/HomeController.cs]: You're right, I have missed this one file during migration.

File: src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Models/AccountViewModels/RegisterViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace DevAdventCalendarCompetition.Models.AccountViewModels
+{
+    public class RegisterViewModel
```
[the remark this answers is not in the document]
Comment (WTobor) [reply src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Models/AccountViewModels/RegisterViewModel.cs]: Ok

**Pull request merged**